### ISSUE TITLE
ReleaseNotesVSTSTask run problem

### ISSUE THREAD

**Issue by bathenme** [Magenic/ReleaseNotesVSTSTask#1]
Hi,
I tried the ReleaseNotesVSTSTask and it is not working for me,
i get this error on my logs:

6-08-28T14:33:43.3898173Z Set workingFolder to default: D:\Agents\Agent3\tasks\BuildReleaseNotes\1.0.6

2016-08-28T14:33:43.9928349Z ##[debug]agent.workFolder=D:\Agents\agent3_work

2016-08-28T14:33:43.9968337Z ##[debug]loading inputs and endpoints

2016-08-28T14:33:43.9978353Z ##[debug]loading ENDPOINT_AUTH_F74646D5-C6A4-40C7-90DF-94E94A06DCF6

2016-08-28T14:33:44.0038332Z ##[debug]loading ENDPOINT_AUTH_SYSTEMVSSCONNECTION

2016-08-28T14:33:44.0048288Z ##[debug]loading INPUT_ADDBUILDNUMBERTOBUGSTASKSFIRSTQUERY

2016-08-28T14:33:44.0048288Z ##[debug]loading INPUT_ADDBUILDNUMBERTOBUGSTASKSSECONDQUERY

2016-08-28T14:33:44.0058273Z ##[debug]loading INPUT_ADDBUILDNUMBERTOBUGSTASKSTHIRDQUERY

2016-08-28T14:33:44.0058273Z ##[debug]loading INPUT_FIRSTQUERYID

2016-08-28T14:33:44.0068292Z ##[debug]loading INPUT_OUTPUTFILE

2016-08-28T14:33:44.0068292Z ##[debug]loading INPUT_TEMPLATEFILE

2016-08-28T14:33:44.0078273Z ##[debug]loaded 8

2016-08-28T14:33:44.4698381Z Uri: http://tfs:8090/tfs/DefaultCollection/

2016-08-28T14:33:44.4708376Z ProjectId: DevOps

2016-08-28T14:33:44.4708376Z SystemVssConnection exists true

2016-08-28T14:33:44.4718362Z ##[debug]SystemVssConnection exists true

2016-08-28T14:33:44.4718362Z connection established

2016-08-28T14:33:44.4718362Z New Working Directory: D:\Agents\agent3_work\77\s

2016-08-28T14:33:44.4748400Z work item interface created.

2016-08-28T14:33:44.4758381Z ##[debug]firstQueryId=6695c27a-00fd-4e5e-b079-cc6f552fafde

2016-08-28T14:33:44.4758381Z ##[debug]addBuildNumberToBugsTasksFirstQuery=false

2016-08-28T14:33:44.4768385Z ##[debug]secondQueryId=null

2016-08-28T14:33:44.4768385Z ##[debug]addBuildNumberToBugsTasksSecondQuery=false

2016-08-28T14:33:44.4778381Z ##[debug]thirdQueryId=null

2016-08-28T14:33:44.4778381Z ##[debug]addBuildNumberToBugsTasksThirdQuery=false

2016-08-28T14:33:44.4778381Z ##[debug]templatefile=\fs\RNTemplateBach.txt

2016-08-28T14:33:44.4788411Z ##[debug]outputfile=D:\Agents\agent3_work\77\a_PublishedWebsites\OpsPortalAgent.WebApi\releasenotesTEST.md

2016-08-28T14:33:44.4878470Z 6695c27a-00fd-4e5e-b079-cc6f552fafde

2016-08-28T14:33:44.5318375Z { [Error: Failed Request: Unauthorized(401) - http://tfs:8090/tfs/DefaultCollection/_apis/wit] statusCode: 401 }

2016-08-28T14:33:44.5318375Z Error with first query: Task failed: Error: Failed Request: Unauthorized(401) - http://tfs:8090/tfs/DefaultCollection/_apis/wit

(run on tfs on premise)

Regards,
Bat-Hen

**Comment by Bowman74** [comment]
@bathenme ,
From the error it looks like the agent the build process is running under does not have permissions to execute your first query. For me this is the "Project Collection Build Services" account. See the instructions under "Setting Up Permissions for the Query": https://github.com/Magenic/ReleaseNotesVSTSTask

Let me know if this helps!

**Comment by bathenme** [comment]
@bowman74, 
Hi,
I alredy add permissions to the user..
And I try to enter to this link whit my user but that link does  not exist..
http://tfs:8090/tfs/DefaultCollection/_apis/wit.....
The project name 'DevOps' is missing or somethinglike this....

**Comment by Bowman74** [comment]
What is the entire URL that you are trying?

**Comment by bathenme** [comment]
The URL that are in the error
http://tfs:8090/tfs/DefaultCollection/_apis/wit

**Comment by Bowman74** [comment]
The unauthorized error, from my experience, means that the user trying to run the query doesn't have permission to run it or that the query with the given Guid doesn't exist.

I'd start my rechecking the query Guid and then if that doesn't work making sure any/all accounts have allow permissions to read the query. We're using VSTS online, I could be possible that on premise installs like yours use a different account.

**Comment by Bowman74** [comment]
@bathenme ,
Did you discover the source of the issue?

**Comment by bathenme** [comment]
@Bowman74 
no..
i have try to check again my GUID and my permissions and everything looks ok.

i try to get into the page http://tfs:8090/tfs/DefaultCollection/_apis/wit with the user that runs the build  and the error is :the page not found. 

its ok? maybe this is my problem ...

**Comment by Bowman74** [comment]
Yes, that would cause it. How are you getting the query Guid, are you running something like this:

https://tfs:8090DefaultCollection/_apis/wit/queries/Shared%20Queries/MyQueryName?api-version=2.2

**Comment by bathenme** [comment]
https://tfs:8090DefaultCollection/projectname/_apis/wit/queries/Shared%20Queries/MyQueryName?api-version=2.2

And I get all the query parameters...

**Comment by lukbud14** [comment]
Anyone fixed this?
